sitemedia/js/controllers/iiif_controller.js

```diff
@@ -4,32 +4,194 @@ import { Controller } from "@hotwired/stimulus";
+            "input[type='checkbox'][id^='zoom-toggle']"
+        );
+        const image = element.querySelector("img.iiif-image");
+        viewer.addHandler("open", () => {
+            // ensure image is positioned in top-left corner of viewer
+            this.resetBounds(viewer);
+            if (isMobile) {
+                // zoom to 110% if on mobile
+                viewer.viewport.zoomTo(1.1);
+            }
+            // initialize zoom slider
+            zoomSlider.setAttribute("min", minZoom);
+            // use toPrecision to ensure no extra pixels on the right of the slider
+            zoomSlider.setAttribute(
+                "max",
+                viewer.viewport.getMaxZoom().toPrecision(2)
             );
+            zoomSlider.addEventListener("input", (evt) => {
+                // Handle changes in the zoom slider
+                let zoom = parseFloat(evt.currentTarget.value);
+                let deactivating = false;
+                if (zoom <= minZoom) {
+                    // When zoomed back out to 100%, deactivate OSD
+                    zoom = minZoom;
+                    this.resetBounds(viewer);
+                    this.deactivateDeepZoom(element, image);
+                    evt.currentTarget.value = minZoom;
+                    deactivating = true;
+                } else {
+                    // Zoom to the chosen percentage
+                    viewer.viewport.zoomTo(zoom);
+                }
+                this.updateZoomUI(
+                    zoom,
+                    deactivating,
+                    zoomSlider,
+                    zoomSliderLabel
+                );
+            });
+            // initialize mobile zoom toggle
+            zoomToggle.addEventListener("input", (evt) => {
+                // always first zoom to 100%
+                viewer.viewport.zoomTo(1.0);
+                if (!evt.currentTarget.checked) {
+                    // wait to reset bounds until element is hidden
+                    setTimeout(() => this.resetBounds(viewer), 300);
+                    this.deactivateDeepZoom(element, image);
+                } else {
+                    // reset bounds immediately and zoom to 110%
+                    this.resetBounds(viewer);
+                    viewer.viewport.zoomTo(1.1);
+                }
+            });
+        });
+        viewer.addHandler("zoom", (evt) => {
+            // Handle changes in the canvas zoom
+            let { zoom } = evt;
+            if (zoom <= minZoom) {
+                // If zooming to less than 100%, force it to 100%
+                zoom = minZoom;
+            }
+            // Set zoom slider value to the chosen percentage
+            zoomSlider.value = parseFloat(zoom);
+            this.updateZoomUI(zoom, false, zoomSlider, zoomSliderLabel);
+        });
+        if (isMobile) {
+            viewer.addHandler("canvas-release", () => {
+                // Handle issue on mobile that causes a scroll trap
```

Comment:
can you elaborate on the scroll trap? Is it if the image is showing mid-screen?

what happens with long images? do we / should we limit them to `100vh` on mobile ?

Reply:
For some reason, on `canvas-release` the OSD container forcibly scrolls into view, which causes the mobile toggle to become hidden. I guess this is to make it easier to manipulate the image, but it makes our control inaccessible.

Long images would be a problem, yes! And the height would likely need to leave space for the toggle control, so maybe something like `100vh - height of toggle` would be a good idea.

Reply:
ah, ok. please elaborate the comment in the code to explain this so we have it documented when we revisit this code.

The height limit you propose sounds good to me. Gissoo may already have some examples of longer images we can test with, or if not we can ask the team.

Reply:
no, haven't run into it yet — I'm good with tracking that on a separate issue so we can move forward